File: longest_streak/longest_streak.js
// class Node {
//   constructor(val) {
//     this.val = val;
//     this.next = null;
//   }
// }
​
const longestStreak = (head) => {
  let prev = null;
  let curr = head;
  let currStreak = 0;
  let maxStreak = 0;
​
  while(curr != null) {
    if(curr.val === prev) {
      currStreak++
      prev = curr.val;
      if(currStreak > maxStreak) maxStreak = currStreak;
    } else {
      currStreak = 1;
      prev = curr.val;
      if(currStreak > maxStreak) maxStreak = currStreak;
    } 
    curr = curr.next;
  }
​
  return maxStreak
};
​
module.exports = {
  longestStreak,
};
​

File: longest_streak/longest_streak.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.next = None
​
def longest_streak(head):
  current = head
  prev = None
  max_streak = 0
  current_streak = 0
​
  while current is not None:
    if current.val == prev:
      prev = current.val
      current_streak += 1
      if current_streak > max_streak:
        max_streak = current_streak
    else:
      prev = current.val
      current_streak = 1
      if current_streak > max_streak:
        max_streak = current_streak
    current = current.next
​
  return max_streak
​